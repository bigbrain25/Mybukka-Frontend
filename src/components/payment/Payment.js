import React, { useEffect } from 'react';

import PropTypes from 'prop-types';
import Modal from 'Components/modal/Modal';
import DismissModal from '../modal/DismissModal';
import { useModalContext } from '../../context/ModalContext';

import PaymentForm from './PaymentForm';
import './Payment.scss';
import Container from '../container/Container';

const PaymentHeader = ({ handleClick }) => (
  <div className="Payment-Form-Header pb-1">
    <div className="text-end">
      <DismissModal onClick={handleClick} />
    </div>
    <div className="Payment-Details-Header">
      <h5 className="Payment-Details-Text">Add Delivery Payment</h5>
    </div>
  </div>
);

const Payment = (props) => {
  const wrapperRef = React.createRef();
  const { paymentPopup, setPaymentPopup, paymentSecurityPopup, setPaymentSecurityPopup, setModal } = useModalContext();

  const handleClickOutside = (event) => {
    if (wrapperRef.current && !wrapperRef.current.contains(event.target)) {
      setModal(false);
      setPaymentPopup(false);
    }
  };

  const requestSecurityPopup = () => {
    if (!props.withModal) setModal(true);
    if (!paymentSecurityPopup) setPaymentSecurityPopup(true);
  };

  const handleClick = (excl) => {
    if (excl && props.withModal) setModal(false);
    setPaymentPopup(false);
  };

  useEffect(() => {
    document.addEventListener('mousedown', handleClickOutside);
    return () => document.removeEventListener('mousedown', handleClickOutside);
  }, [wrapperRef]);

  const paymentJsx = <PaymentForm requestSecurityPopup={requestSecurityPopup} handleClick={handleClick} {...props} />;

  if (props.withModal) {
    return (
      <Modal show={paymentPopup} bodyClassName="SmallWidth" ref={wrapperRef}>
        <Container classNames="Payment-Wrapper">
          <PaymentHeader handleClick={() => handleClick(true)} />
          {paymentJsx}
        </Container>
      </Modal>
    );
  }
  return paymentJsx;
};

export default Payment;

Payment.defaultProps = {
  type: 'payment'
};

Payment.propTypes = {
  type: PropTypes.string
};
