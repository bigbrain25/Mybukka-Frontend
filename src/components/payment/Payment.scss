.Payment-Wrapper {
  max-width: 600px;
}

.Payment-Checkox--text {
  margin-left: 5px;
  font-size: 12px;
}

.Payment-Checkox-Wrapper {
  align-items: center;
  text-align: center;
  color: rgb(143, 149, 163);
  display: flex;
  margin-bottom: 30px;
  // border-bottom: 1px solid rgb(236, 237, 239);
  padding: 30px 0px 24px;
}

.Payment-Details-Header {
  border-bottom: 1px solid #eceff1;
  padding-bottom: 10px;
  width: 100%;
}

.Payment-Details-Text {
  font-weight: 500;
}

.Payment-Form-Header {
  display: flex;
  flex-direction: column;
  width: 100%;
  padding-top: 20px;
}

.text-end {
  text-align: end
}

@media (max-width: 991.98px) {
  .add-card-form {
    height: 100vh;
    width: 100%;
  }
}
  