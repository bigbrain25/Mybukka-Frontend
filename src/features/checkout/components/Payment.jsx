import React, { useState, useEffect } from 'react';

import { connect } from 'react-redux';
import swal from 'sweetalert';

import manipulateCardDetailsAction from 'Redux/manipulateCardDetailsAction';
import getUserCard from '../actionCreators/getUserCard';

import { validateAFieldPayment, validateAllFieldsPayment } from '../validation/validateField';
import Demarcation from '../common/SmallScreenDivider';
import Card, { AddCard } from './Card';
import setDefaultCard from '../actionCreators/setDefaultCard';

import './payment.scss';

const Payment = ({
  manipulateCardDetails,
  handleClick,
  message,
  fetchUserCard,
  cards,
  setCardAsDefault,
  status,
  isCardSaved,
}) => {
  const [active, setActive] = useState(false);
  const [addCard, showCardForm] = useState(false);
  const [validationErrors, setValidationErrors] = useState({
    number: '',
    expDate: '',
    cvv: '',
  });

  const [inputData, setInputData] = useState({
    number: '',
    expDate: '',
    cvv: '',
  });

  const handleChange = ({ target: { name, value } }) => {
    const newFieldData = { [name]: value };
    const validation = validateAFieldPayment(newFieldData, name);
    setInputData({
      ...inputData,
      ...newFieldData
    });
    setValidationErrors({
      ...validationErrors,
      [name]: validation.message
    });
  };

  const manipulateCard = () => {
    const { expDate, cvv, number } = inputData;
    const cardValidityDates = expDate ? expDate.split('/') : ['', ''];
    return {
      cvv,
      number: number.split(' ').join(''),
      expiry_month: cardValidityDates[0],
      expiry_year: cardValidityDates[1],
    };
  };

  const handleSaveButton = (event) => {
    event.preventDefault();
    const { passes, errors } = validateAllFieldsPayment(inputData);

    setValidationErrors({
      ...validationErrors,
      ...errors
    });
    if (passes && !active) {
      manipulateCardDetails(inputData);
      handleClick({ card: { ...manipulateCard() }, amount: 100 });
      setActive(true);
    }
  };

  const handleDefaultSelection = async (id) => {
    await setCardAsDefault(id);
    await fetchUserCard();
  };

  useEffect(() => {
    showCardForm(false);
    fetchUserCard();
  }, [isCardSaved]);

  useEffect(() => {
    fetchUserCard();
  }, []);

  useEffect(() => {
    setActive(false);
  }, [status]);

  return (
    <section className="mb-2 mt-4">
      <h2 className="font-size-16 px-3 px-md-3 px-lg-0">Payment</h2>
      <form className="border padding-20 mt-4" action="">
        {!addCard && cards.length > 0 ?
          <div>
            <p>Your credits card</p>
            { cards.map(card => (
              <Card
                handleClick={() => handleDefaultSelection(card._id)} // eslint-disable-line
                selected={card.selected}
                last4={card.last4}
                expiredYear={card.exp_year}
                expiredMonth={card.exp_month}
                cardType={card.card_type}
              />))}
            <div
              onKeyDown={() => {}}
              role="button"
              aria-pressed="false"
              tabIndex="0"
              onClick={() => showCardForm(true)}
              className="text-muted cursor-pointer"
            >+ add card</div>
          </div>
          : <AddCard
            handleChange={handleChange}
            cards={cards}
            inputData={inputData}
            active={active}
            handleSaveButton={handleSaveButton}
            handleClick={() => showCardForm(false)}
            validationErrors={validationErrors}
          />
        }
        {/* {message === 'Charge attempted' &&
        <p className="text-success">Your card is now saved for any other transactions</p>} */}
      </form>
      <Demarcation />
    </section>
  );
};

const mapStateToProps = ({
  getUserCardReducer: { cards },
  chargeUserReducer: {
    data: { status }
  },
  saveUserCardReducer: {
    newPayment: { card: isCardSaved },

  },
}) => ({
  cards,
  status,
  isCardSaved,
});

export default connect(
  mapStateToProps,
  { manipulateCardDetails: manipulateCardDetailsAction,
    fetchUserCard: getUserCard,
    setCardAsDefault: setDefaultCard,
  }
)(Payment);

