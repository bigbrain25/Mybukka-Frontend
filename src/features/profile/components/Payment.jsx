import React from 'react';

import { BasicPlus } from 'Icons/Plus';

import AddMoreSection from '../common/AddMoreSection';
import AccountDetailsGroupHeader from '../common/AccountDetailsGroupHeader';
import AddPaymentForm from '../forms/AddPaymentForm';

import './payment.scss';

const AddPayment = () => (
  <div className="add-payment">
    <AddMoreSection text="Add payment card" dataTarget="#modal">
      <BasicPlus />
    </AddMoreSection>
  </div>
);

const Payment = () => (
  <div className="payment-details-section">
    <AddPaymentForm type="payment" />
    <AccountDetailsGroupHeader text="payment" />
    <AddPayment />
  </div>
);

export default Payment;
