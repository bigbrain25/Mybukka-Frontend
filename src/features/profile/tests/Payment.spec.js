import React from 'react';
import { Provider } from 'react-redux';
import Payment from '../components/Payment';

const store = mockStore({
  deliveryModeReducer: {
    mode: 'pickup'
  },
  updateUserAddressReducer: { errorMessage: '', posted: false },
  selectedLocationReducer: { coordinates: [3.361476, 6.5560715] }
});

describe('PaymentSection Component', () => {
  const wrapper = mount(
    <Provider store={store}>
      <Payment />
    </Provider>
  );

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
