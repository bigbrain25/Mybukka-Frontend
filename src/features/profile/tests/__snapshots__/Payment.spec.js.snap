// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentSection Component renders properly 1`] = `ReactWrapper {}`;
